privx_api/base.py

```diff
@@ -12,7 +12,14 @@
 
 
 def format_path_components(format_str: str, **kw) -> str:
-    components = {k: urllib.parse.quote(v, safe="") for k, v in kw.items()}
+    try:
+        components = {k: urllib.parse.quote(v, safe="") for k, v in kw.items()}
+    except TypeError:
+        incorrect_params = {
+            k: f"{type(v)}" for k, v in kw.items() if isinstance(v, str) is False
+        }
+        error_message = f"Incorrect ...:\n{json.dumps(incorrect_params, indent=4)}"
```

Comment:
The example code was just for discussion purpose. It should not be taken as is. The error message should not contain ... in this case. Let's come up with a precise error description.